src/services/rollouts.js

```diff
@@ -540,4 +543,216 @@ module.exports = class RolloutsHelper {
 			return error
 		}
 	}
+	/**
+	 * rollout publish
+	 * @method
+	 * @name publish
+	 * @param {Integer} rolloutId - rollout id
+	 * @returns {JSON} - rollout publish response.
+	 */
+
+	static async publish(rolloutId, orgId, loggedInUserId) {
+		try {
+			// fetch rollout details
+			const rolloutDetails = await this.details(rolloutId, orgId, loggedInUserId)
+
+			// check if rollout is present or not
+			if (rolloutDetails?.statusCode != 200 || rolloutDetails?.result == undefined) return rolloutDetails
+
+			if (rolloutDetails?.result?.status && rolloutDetails?.result?.status == common.ROLLOUT_PUBLISHED) {
+				return responses.failureResponse({
+					responseCode: 'CLIENT_ERROR',
+					statusCode: httpStatusCode.bad_request,
+					result: [],
+					message: 'ROLLOUT_ALREADY_PUBLISHED',
+				})
+			}
+
+			const validateRollout = await this.validateRollout(rolloutDetails?.result)
+			if (validateRollout.length > 0) {
+				const result = Array.isArray(validateRollout) ? validateRollout.flat() : validateRollout || []
+				return responses.failureResponse({
+					responseCode: 'CLIENT_ERROR',
+					statusCode: httpStatusCode.bad_request,
+					result: result,
+					message: 'ROLLOUT_VALIDATION_FAILED',
+				})
+			}
+
+			//remove after fixing kafka issue
+			await this.publishCallback(rolloutId, null)
+
+			// fetch resource details
+			const resourceDetails = await resourceService.getDetails(rolloutDetails?.result?.resource_id, orgId)
+			// check if resource is present or not
+			if (resourceDetails?.statusCode != 200 || resourceDetails?.result == undefined) return resourceDetails
+			// publish the resource if not published
+			if (
+				resourceDetails?.result?.status != common.RESOURCE_STATUS_PUBLISHED ||
```

Comment:
@adithyadinesh0412 where are we handling the rollout update after publish, that also will call the same api, so that time it should not duplicate solution and program